src/dmd/globals.d

```diff
@@ -362,69 +362,81 @@ struct Global
 
     extern (C++) void _init()
     {
-        static if (TARGET.Windows)
+        _version = import("VERSION") ~ '\0';
+
+        version (MARS)
         {
-            obj_ext = "obj";
+            vendor = "Digital Mars D";
+            static if (TARGET.Windows)
+            {
+                obj_ext = "obj";
+            }
+            else static if (TARGET.Linux || TARGET.OSX || TARGET.FreeBSD || TARGET.OpenBSD || TARGET.Solaris || TARGET.DragonFlyBSD)
+            {
+                obj_ext = "o";
+            }
+            else
+            {
+                static assert(0, "fix this");
+            }
+            static if (TARGET.Windows)
+            {
+                lib_ext = "lib";
+            }
+            else static if (TARGET.Linux || TARGET.OSX || TARGET.FreeBSD || TARGET.OpenBSD || TARGET.Solaris || TARGET.DragonFlyBSD)
```

Comment:
This condition is exactly the same as `obj_ext`, isn't it ? Can you just merge them together ?
And it would be much more readable if we had:
```d
static if (TARGET.Windows)
{
    params.mscoff = params.is64bit;
    // More stuff
}
else static if (TARGET.Linux || TARGET.OSX || ...)
{
    static if (TARGET.OSX)
      dll_ext = "dylib";
    else
      dll_ext = "so";
}
```

Instead of having a block for everything

Reply:
It is tempting to refactor, though discouraged to do two things in one PR?